django/contrib/contenttypes/fields.py

```diff
@@ -616,9 +648,17 @@ def get_queryset(self):
                 return self._apply_rel_filters(queryset)
 
         def get_prefetch_queryset(self, instances, queryset=None):
+            warnings.warn(
+                "Calling get_prefetch_queryset() is deprecated. "
+                "Use get_prefetch_querysets() instead.",
+                RemovedInDjango60Warning,
+            )
             if queryset is None:
-                queryset = super().get_queryset()
+                return self.get_prefetch_querysets(instances)
+            return self.get_prefetch_querysets(instances, [queryset])
 
+        def get_prefetch_querysets(self, instances, querysets=None):
+            queryset = querysets[0] if querysets is not None else super().get_queryset()
```

Comment:
I would raise a `ValueError` when `len(queryset) > 1`.

Reply:
The behaviour of `get_prefetch_queryset` is duplicated 5 times and adding the length check would be also duplicated. Does it make sense to create a util method somewhere or is duplicating this behaviour (which will be removed anyway) fine?

Reply:
The `get_prefetch_queryset` will be removed no? (This is the method that is marked as deprecated and that is also duplicated everywhere)

Reply:
> The `get_prefetch_queryset` will be removed no? (This is the method that is marked as deprecated and that is also duplicated everywhere)

We are inside `get_prefetch_querysets()`, not in `get_prefetch_queryset()`.

Reply:
OK 👍